lang/luasocket/Makefile

```diff
@@ -8,33 +8,59 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luasocket
-PKG_SOURCE_VERSION:=6d5e40c324c84d9c1453ae88e0ad5bdd0a631448
-PKG_VERSION:=3.0-rc1-20130909
-PKG_RELEASE:=5
+PKG_SOURCE_DATE:=2019-04-21
+PKG_SOURCE_VERSION:=733af884f1aa18ff469bf3c4d18810e815853211
+PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
-PKG_MIRROR_HASH:=d2fa075d8bd026c41e0eb1a634ac2ad8115dee8abb070720e8e91fab51f86ee4
-PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket
+PKG_MIRROR_HASH:=60aef7544426cae3e6c7560a6e4ad556a04b879ca0ad0311645b2c513c872128
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_SOURCE_DATE)-$(call version_abbrev,$(PKG_SOURCE_VERSION))
+
+PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/luasocket
+define Package/luasocket/default
   SUBMENU:=Lua
   SECTION:=lang
   CATEGORY:=Languages
+  URL:=http://w3.impa.br/~diego/software/luasocket
+endef
+
+define Package/luasocket
+  $(Package/luasocket/default)
   TITLE:=LuaSocket
-  URL:=http://luasocket.luaforge.net/
-  MAINTAINER:=W. Michael Petullo <mike@flyn.org>
   DEPENDS:=+lua
+  VARIANT:=five-one
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/luasocket-5.3
+  $(Package/luasocket/default)
+  TITLE:=LuaSocket 5.3
+  DEPENDS:=+liblua5.3
```

Comment:
as above, I'd personally like to see things be as consistent as possible wrt version suffices

Reply:
@karlp - I'm sorry, I don't understand what you're asking for.

Reply:
@cwluxul: we have `fooN.M`, so don't start using `foo-N.M` please. Make that `luasocket5.3`.

Reply:
@rmilecki - I changed from luasocket-5.3 to luasocket5.3.  This will no longer build the variant.  I looked around at other makefiles with VARIANT, the variant defines all have dashes (which I'm guessing is how I ended up with a dash to begin with, trying to mirror what has already been done).  I think the dash must be part of the build system for variants?

Reply:
Oh, I did make sure to update the feeds and modify the .config to reflect the new package name, BTW.

Reply:
@cwluxul: I took your commit, renamed `luasocket-5.3` to `luasocket5.3` and it still compiles for me. What exact error did you see?

Reply:
@rmilecki - There was not an error, the variant simply didn't build.  But I just re-made that change to test again and today it worked.  I must have been making a mistake last week.  I've updated. this.  @karlp

Reply:
@rmilecki - There was not an error, the variant simply didn't build.  But I just re-made that change to test again and today it worked.  I must have been making a mistake last week.  I've updated. this.  @karlp